src/app/userDashboard/services/crud.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable, inject } from '@angular/core';
+import {
+  Firestore,
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDocs,
+} from '@angular/fire/firestore';
+import { ITrip } from '../models/trip';
+import { Observable, from } from 'rxjs';
+import { environment } from '../../../environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CrudService {
+  firestore = inject(Firestore);
+
+  addTrip(
+    userEmail: string,
```

Comment:
I feel like you can make email, tripName and destination into an interface as you reuse it so much

Reply:
It is part of the Trip interface but the reason i pass it here like this is because I only want to 'assemble' a trip object once i get it from the Firestore. This is cause I don't want to store derivable information in the database. So for this part I only want to use the userEmail for the path and then the tripName and destination to store in the database. So since I would deconstruct the interface anyways, I thought i would introducing unnecessary overhead by passing it as an interface here.